x-pack/platform/plugins/shared/actions/common/validate_email_addresses.ts

```diff
@@ -14,8 +15,20 @@ import { hasMustacheTemplate } from './mustache_template';
 export interface ValidateEmailAddressesOptions {
   /** treat any address which contains a mustache template as valid */
   treatMustacheTemplatesAsValid?: boolean;
+  // addresses with this option won't be validated against the allowed recipient list
+  isSender?: boolean;
 }
 
+export const isAddressMatchingSomePattern = (address: string, patterns: string[]): boolean => {
+  return patterns.some((pattern) => {
+    // Escape special regex characters in the pattern except for '*'
+    const regexStr = '^' + pattern.split('*').map(escapeRegExp).join('.*') + '$';
+    const regex = new RegExp(regexStr, 'i');
```

Comment:
I did use case-insensitive matching because even though the RFC states [The local-part of a mailbox MUST BE treated as case sensitive.](https://www.ietf.org/rfc/rfc5321.txt), most email providers ignore this and because the domain part is case-insensitive. However, this might be an issue if the customer uses an email service with case-sensitivity and makes use of it.

Reply:
This seems fine.  Would only be a problem for an email system with "user" names that only differed in case?